File: frontend/src/api/support.ts
import { api } from "./client";

export interface SupportTurn {
  role: "user" | "assistant";
  content: string;
}

export const supportApi = {
  /** Ask the AI assistant. `history` is the prior turns (excluding `message`). */
  chat: (message: string, history: SupportTurn[]) =>
    api<{ reply: string; source: "ai" | "unavailable" }>("/support/chat", {
      method: "POST",
      body: { message, history },
    }),

  /** "Talk to a human" — emails the support inbox. */
  escalate: (message: string, history: SupportTurn[]) =>
    api<{ ok: boolean; error?: string }>("/support/escalate", {
      method: "POST",
      body: { message, history },
    }),
};

File: backend/schemas/support.py
from typing import Literal, Optional
from pydantic import BaseModel, Field


class SupportTurn(BaseModel):
    role: Literal["user", "assistant"]
    content: str = Field(min_length=1, max_length=2000)


class SupportChatSchema(BaseModel):
    """A user message to the support assistant, with optional prior turns."""
    message: str = Field(min_length=1, max_length=2000)
    history: Optional[list[SupportTurn]] = None


class SupportEscalateSchema(BaseModel):
    """A 'talk to a human' request — persisted as a ticket + emailed to the inbox."""
    message: str = Field(min_length=1, max_length=4000)
    history: Optional[list[SupportTurn]] = None


class AdminUpdateTicketSchema(BaseModel):
    """An admin resolving (or reopening) a support ticket from the desk."""
    status: Literal["open", "closed"]
    note: Optional[str] = Field(default=None, max_length=1000)
